cmd/check.go

```diff
@@ -170,31 +157,34 @@ func checkDBM(config *internal.Config, wd string) error {
 		return fmt.Errorf("failed to parse model: %w", err)
 	}
 
-	modelRoles := map[string]struct{}{}
+	modelRoles := map[string]dbm.Role{}
 	for _, role := range model.Roles {
-		if !role.SQLDisabled {
-			//nolint:err113
-			return fmt.Errorf("role %q has sql enabled", role.Name)
-		}
-		modelRoles[role.Name] = struct{}{}
+		modelRoles[role.Name] = role
 	}
 
-	configRoles := map[string]struct{}{}
-	for _, role := range config.DatabaseUsers {
-		configRoles[role] = struct{}{}
+	configRoles := map[string]configuration.Role{}
+	for _, role := range config.Roles {
+		configRoles[role.Name] = role
 	}
 
 	for role := range modelRoles {
 		if _, ok := configRoles[role]; !ok {
 			//nolint:err113
-			return fmt.Errorf("role %q defined in the model not defined in the config", role)
+			return fmt.Errorf("role %q is defined in the model but is not defined in the config", role)
 		}
 	}
 
 	for role := range configRoles {
 		if _, ok := modelRoles[role]; !ok {
 			//nolint:err113
-			return fmt.Errorf("role %q defined in the config not defined in the model", role)
+			return fmt.Errorf("role %q is defined in the config but is not defined in the model", role)
+		}
+	}
+
+	for _, role := range model.Roles {
+		if !role.SQLDisabled {
+			//nolint:err113
+			return fmt.Errorf("role %q is missing 'sql disabled' in the model (sql must not be generated for a role)", role.Name)
 		}
 	}
```

Comment:
From the other changes I understood that the roles in the config are now externally managed, so `trek apply` expects them to exist already and `trek check`/`trek generate` will create them automatically. This check contradicts this, but maybe I understood it wrong.

Checking for sql-disabled should also not be necessary anymore, although we could at least check for all roles with sql-disabled=true that they are listed in the config file.

Reply:
Discussed in chat. For future readers:
- sql-disables must still be enabled. roles must be externally managed.
- check/generate will only create them on the database trek starts (manages) itself.